func/code.func

```diff
@@ -90,107 +92,127 @@ _ unpack_order(slice order) inline_ref {
       return ();
   }
 
+  (slice stats, cell orders, slice owner, slice ecvrf, slice player, int player_coins) = unpack_state();
   slice sender = cs~load_msg_addr();
 
-  (slice stats, cell orders, slice owner1, slice owner2, slice echo) = unpack_state();
-
-  if (slices_equal?(sender, owner1) | slices_equal?(sender, owner2)) {
+  if (slices_equal?(sender, owner)) {
     ;; owners can only deposit
     return ();
   }
 
-  if (slices_equal?(sender, echo)) {
-    (slice player, int o_amount) = (in_msg_body~load_msg_addr(), in_msg_body~load_coins());
-    ;; lottery inactive if balance < 2 stake + 1 TON
-    if (my_tons() < o_amount * 2 + 1000000000) {
-       send_tons(player, msg_value, 2);
-    } else {
-      (int o_seqno, int n_wins, int in_amount, int out_amount, int hash) = unpack_stats(stats);
-      ;; Calculating new hash
-      hash = cell_hash(begin_cell()
-        .store_uint(hash, 256)
-        .store_uint(msg_hash, 256)
-        .store_uint(cur_lt(), 64)
-        .store_uint(now(), 64)
-        .end_cell()
-      );
-      randomize(hash);
-      if (rand(10) >= 4) {
-        builder order = pack_order(o_seqno, 3, now(), o_amount, player);
-        orders~udict_set_builder(64, o_seqno, order);
-        in_amount += o_amount;
-        if (o_amount > 3000000000) {
-            send_tons(owner1, o_amount / 6, 3);
-            send_tons(owner2, o_amount / 6, 3);
-        }
-      } else {
-        builder order = pack_order(o_seqno, 2, now(), o_amount, player);
-        orders~udict_set_builder(64, o_seqno, order);
-
-        in_amount += o_amount;
-        out_amount += o_amount * 2;
-        n_wins += 1;
-
-        send_tons(player, 2 * o_amount, 3);
+  int op = in_msg_body~load_uint(32);
+
+  ;; new random number received from vrf
+  if (op == 0x069CECA8 & slices_equal?(sender, ecvrf)) {
```

Comment:
Can have an issue with operators priority if & is executed before ==.

Reply:
Will changing it to this help? -` if ((op == 0x069CECA8) & (slices_equal?(sender, ecvrf)))`